pkg/hnsw/hnsw.go

```diff
@@ -145,3 +162,32 @@ func (h *Hnsw) selectNeighbors(candidates *MinQueue, numNeighborsToReturn int) *
 
 	return nil
 }
+
+func (h *Hnsw) KnnSearch(q Vector, kNeighborsToReturn, ef int) ([]*Item, error) {
+	currentNearestElements := NewMinQueue()
+	entryPointNode := h.Nodes[h.EntryNodeId]
+
+	for l := entryPointNode.layer; l >= 1; l-- {
+		numNearestToQAtLevelL := h.searchLayer(q, entryPointNode, 1, l)
+
+		for !numNearestToQAtLevelL.IsEmpty() {
+			peeled := numNearestToQAtLevelL.Peel()
+			currentNearestElements.Insert(peeled.id, peeled.dist)
+		}
```

Comment:
for a given layer, get a minqueue of the nearest neighbors nearest to q. we have to add it to the currentNearestElements.

Is there a better way to do this? Maybe encapsulating this logic inside `queue` class

Reply:
I think this is ok, probably some kind of bulk insert op would be better but it's not a huge deal because there might not be that much improvement with the heap.